cpp/mcap/include/mcap/writer.inl

```diff
@@ -69,6 +69,20 @@ void FileWriter::flush() {
 
 void FileWriter::end() {
   if (file_) {
+    std::fflush(file_);
+#if defined(_WIN32) || defined(_WIN64)
```

Comment:
we have a different set of platform-detection ifdefs in `visibility.hpp` - we don't check for win64, we do check for cygwin. 

From https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170, it looks like `_WIN64` is redundant with `_WIN32`.

Also, `::fsync` is only defined in `unistd.h`, which we don't include in `mcap`, so this code only works when built into another object that includes it. We probably need a conditional include at the top of the file.

Reply:
Similar with `_fileno`, which microsoft says is defined in `stdio.h` - we only include `cstdio`, which may resolve to the same thing? I haven't tested. https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fileno?view=msvc-170